pkg/logs/sender/destination_sender.go

```diff
@@ -46,7 +46,10 @@ func (d *DestinationSender) startRetryReader() {
 		for v := range d.retryReader {
 			d.retryLock.Lock()
 			if d.cancelSendChan != nil && !d.lastRetryState {
-				d.cancelSendChan <- struct{}{}
+				select {
+				case d.cancelSendChan <- struct{}{}:
```

Comment:
I already doubt about this but I can't remember the conclusion: are you sure that the default case is always happening after a test of the others? I vaguely remember that yes but I'm not sure. Because otherwise, this channel will never be emptied (not that it matters a lot...).

Reply:
Yes, default is executed iff other branches are not ready. This channel acts like a flag: if it's empty, we add an item. If not empty, it will interrupt Send as is, and we don't need to do anything. `Send` creates a fresh channel and drops it on return, so any out of date messages will be dropped as well.

Reply:
💯 
Thank you!